crates/ere-hosts/src/main.rs

```diff
@@ -122,56 +83,30 @@ impl From<BenchmarkAction> for Action {
 }
 
 /// Main entry point for the host benchmarker
-#[tokio::main]
-async fn main() -> Result<(), Box<dyn std::error::Error>> {
+fn main() -> Result<(), Box<dyn std::error::Error>> {
     let cli = Cli::parse();
 
     let resource: ProverResourceType = cli.resource.into();
     let action: Action = cli.action.into();
 
-    let block_witness_gen: Box<dyn WitnessGenerator> = match cli.source {
-        SourceCommand::Tests {
-            directory_path,
-            include,
-            exclude,
-        } => Box::new(ExecSpecTestBlocksAndWitnesses::new(
-            directory_path,
-            include.unwrap_or_default(),
-            exclude.unwrap_or_default(),
-        )),
-        SourceCommand::Rpc {
-            last_n_blocks,
-            block,
-            rpc_url,
-            rpc_header,
-        } => {
-            let parsed_headers: Vec<(String, String)> = rpc_header
-                .unwrap_or_default()
-                .into_iter()
-                .map(|header| {
-                    header
-                        .split_once(':')
-                        .map(|(k, v)| (k.to_string(), v.to_string()))
-                        .ok_or_else(|| {
-                            format!("invalid header format: '{}'. expected 'key:value'", header)
-                        })
-                })
-                .collect::<Result<_, _>>()?;
-
-            let mut builder =
-                RPCBlocksAndWitnessesBuilder::new(rpc_url).with_headers(parsed_headers)?;
-
-            if let Some(block_num) = block {
-                builder = builder.block(block_num);
+    println!("Loading corpuses from: {}", cli.input_folder.display());
+    let corpuses = WalkDir::new(&cli.input_folder)
+        .min_depth(1)
+        .into_iter()
+        .collect::<Result<Vec<_>, _>>()?
+        .into_par_iter()
+        .map(|entry| {
+            if entry.file_type().is_file() {
+                let content = std::fs::read(entry.path())?;
+                let blocks_and_witnesses = serde_json::from_slice(&content).map_err(|e| {
+                    anyhow::anyhow!("Failed to parse {}: {}", entry.path().display(), e)
+                })?;
+                Ok(blocks_and_witnesses)
             } else {
-                builder = builder.last_n_blocks(last_n_blocks.unwrap_or(1));
+                anyhow::bail!("Invalid input folder structure: expected files only")
```

Comment:
I prefered to go strict and don't allow any sub-folder or similar, since that should never happen.